samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/samples/FileTransferIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.findUnallocatedServerPort;
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.propOrEnv;
+
+
+/**
+ * System Test for Sample 04.0-file-transfer
+ */
+@Tag("SystemTests")
+public class FileTransferIntegrationTest {
+
+    private static final String PROVIDER_ASSET_NAME = "test-document";
+
+    private static final String CONTRACT_NEGOTIATION_PATH = "/api/negotiation";
+    private static final String CONTRACT_AGREEMENT_PATH = "/api/control/negotiation/{contractNegotiationRequestId}";
+    private static final String TRANSFER_PATH = "/api/transfer/{transferId}";
+    private static final String FILE_TRANSFER_PATH = "/api/file/{filename}";
+
+    private static final String CONNECTOR_ADDRESS_PARAM = "connectorAddress";
+    private static final String CONTRACT_NEGOTIATION_REQUEST_ID_PARAM = "contractNegotiationRequestId";
+    private static final String TRANSFER_ID_PARAM = "transferId";
+    private static final String DESTINATION_PARAM = "destination";
+    private static final String CONTRACT_ID_PARAM = "contractId";
+    private static final String FILE_NAME_PARAM = "filename";
+
+    private static final String CONSUMER_ASSET_PATH = propOrEnv("edc.samples.04.consumer.asset.path", tempDirectory());
+    private static final int CONSUMER_CONNECTOR_PORT = findUnallocatedServerPort();
+    private static final String CONSUMER_CONNECTOR_HOST = "http://localhost:" + CONSUMER_CONNECTOR_PORT;
+
+    private static final int PROVIDER_CONNECTOR_PORT = findUnallocatedServerPort();
+    private static final String PROVIDER_CONNECTOR_HOST = "http://localhost:" + PROVIDER_CONNECTOR_PORT;
+    private static final String PROVIDER_ASSET_PATH = propOrEnv("edc.samples.04.asset.path", format("%s/%s.txt", tempDirectory(), PROVIDER_ASSET_NAME));
+
+    private static final String API_KEY_CONTROL_AUTH = propOrEnv("edc.api.control.auth.apikey.value", "password");
+    private static final String API_KEY_HEADER = "X-Api-Key";
+
+    @RegisterExtension
+    static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer",
+            Map.of(
+                    "web.http.port", String.valueOf(CONSUMER_CONNECTOR_PORT),
+                    "edc.api.control.auth.apikey.value", API_KEY_CONTROL_AUTH,
+                    "ids.webhook.address", CONSUMER_CONNECTOR_HOST));
+
+    @RegisterExtension
+    static EdcRuntimeExtension provider = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:provider",
+            Map.of(
+                    "web.http.port", String.valueOf(PROVIDER_CONNECTOR_PORT),
+                    "edc.samples.04.asset.path", PROVIDER_ASSET_PATH,
+                    "ids.webhook.address", PROVIDER_CONNECTOR_HOST));
+
+    @BeforeAll
+    static void setUp() {
+        // Consumer connector host URI.
+        RestAssured.baseURI = CONSUMER_CONNECTOR_HOST;
```

Comment:
Setting static stuff on test tools is pretty dangerous since that will involve other tests that use restassured and lead to unexpected failures. Better specify it on every call in my opinion.